src/emqx_broker.erl

```diff
@@ -239,7 +239,7 @@ dispatch(Topic, Delivery = #delivery{message = Msg, results = Results}) ->
 
 dispatch({SubPid, _SubId}, Topic, Msg) when is_pid(SubPid) ->
     SubPid ! {dispatch, Topic, Msg};
-dispatch({share, _Group, _Sub}, _Topic, _Msg) ->
```

Comment:
I suggest delete this clause as there should be no '{share, x, x }' in emqx_subscriber table.